### PR TITLE
chore: Update pull request template

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,8 +1,4 @@
 Checklist:
 
-* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
-* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
 * [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
-* [ ] I've signed the CLA.
```
Comment (alexec) [inline .github/pull_request_template.md:6]: not needed the DCO

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,8 +1,4 @@
...
-* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
```
Comment (alexec) [inline .github/pull_request_template.md:7]: everyone seems to ignore this

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,8 +1,4 @@
 Checklist:
 
-* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
-* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.
```
Comment (alexec) [inline .github/pull_request_template.md:4]: enforced by check

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,8 +1,4 @@
 Checklist:
 
-* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
```
Comment (alexec) [inline .github/pull_request_template.md:3]: not needed in many cases

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,8 +1,4 @@
...
-* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
 * [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
```
Comment (alexec) [inline .github/pull_request_template.md:3]: I think this is effective - keep it

**Review by whynowy — APPROVED** [review]

**Pull request merged**